lib/declarative/option.rb

```diff
@@ -20,13 +20,24 @@ def callable?(value, options)
       value.is_a?(options[:callable] || Callable)
     end
 
-    def lambda_for_proc(value, options)
-      return ->(context, *args) { context.instance_exec(*args, &value) } if options[:instance_exec]
-      value
-    end
-
-    def lambda_for_symbol(value, options)
-      ->(context, *args){ context.send(value, *args) }
+    if RUBY_VERSION >= '2.7.0'
```

Comment:
Can we use `ruby2_keywords` instead of this check ? It'll warn us in future when it's going to get deprecated and we remember to remove it.

Reply:
Thank you for your response. I'll update this.

Reply:
@yogeshjain999: I tried adding `ruby2_keywords` like this:

```ruby
    def lambda_for_proc(value, options)
      return ->(context, *args) { context.instance_exec(*args, &value) } if options[:instance_exec]
      value
    end
    ruby2_keywords :lambda_for_proc if respond_to?(:ruby2_keywords)

    def lambda_for_symbol(value, options)
      ->(context, *args){ context.send(value, *args) }
    end
    ruby2_keywords :lambda_for_symbol if respond_to?(:ruby2_keywords)
```

and the tests failed below cases for ruby 3:

```
  1) Failure:
callable: Object#test_0003_anonymous [/Users/hieunguyen/code/declarative-option/test/option_test.rb:58]:
Expected: "Hello! [1] {:a=>2}"
  Actual: "Hello! [1, {:a=>2}] {}"

  2) Failure:
OptionTest#test_0015_anonymous [/Users/hieunguyen/code/declarative-option/test/option_test.rb:71]:
Expected: "Hello! [1] {:a=>2}"
  Actual: "Hello! [1, {:a=>2}] {}"

  3) Failure:
OptionTest#test_0010_anonymous [/Users/hieunguyen/code/declarative-option/test/option_test.rb:29]:
Expected: "Hello! [1, 2] {:a=>3}"
  Actual: "Hello! [1, 2, {:a=>3}] {}"
```

I wonder what is the correct behavior here?

Reply:
oh yeah, accepting `**options` with an empty hash behaves differently in ruby < 2.6, ruby-2.7 and ruby-3, very confusing.